### PR TITLE
Select2: Tweak - New button for adding a new option to the select box.

### PR DESCRIPTION
**Pull request by rotemee**
## PR Checklist

- [ ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

### CODE REVIEW

File: assets/lib/e-select2/js/e-select2.full.js [hunk 1]
```diff
@@ -3240,10 +3240,12 @@ S2.define('select2/data/select',[
 
     container.on('select', function (params) {
       self.select(params.data);
+      container._setAllSelectedClass();
```
Comment (kobizz) [inline assets/lib/e-select2/js/e-select2.full.js:3243]: We have to find a way to do it as an external extension for select2

**Review by kobizz — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/utils/select2.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+	}
+
+	onDestroy() {
+		super.onDestroy();
+
+		this.destroy();
+	}
+
+	onInit( ...args ) {
+		super.onInit(...args);
```
Comment (KingYes) [inline assets/dev/js/editor/utils/select2.js:58]: Fix JSlint here..

**Review by KingYes — CHANGES_REQUESTED** [review]

**Review by kobizz — CHANGES_REQUESTED** [review]

File: assets/dev/scss/editor/_select2.scss [hunk 3]
```diff
@@ -47,6 +47,18 @@
 			border-color: $editor-light;
 		}
 	}
+
+	&:not(.select2-container--all-selected) .select2-selection--multiple .select2-search--inline::before {
```
Comment (kobizz) [inline assets/dev/scss/editor/_select2.scss:51]: Add to the `.select2-search--inline` class a `padding-top: 0.6px` to place it in the accurate location

**Review by kobizz — CHANGES_REQUESTED** [review]

File: assets/dev/scss/editor/_select2.scss [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+	&.select2-container--open,
+	&.select2-container--focus {
+
+		.select2-selection--single,
+		.select2-selection--multiple {
+			border-color: $editor-light;
+		}
+	}
+
+	& .select2-selection--multiple .select2-search--inline {
```
Comment (kobizz) [inline assets/dev/scss/editor/_select2.scss:51]: Remove the `&` character and add a comment to explain why this line is needed

**Review by kobizz — CHANGES_REQUESTED** [review]

File: assets/dev/scss/editor/_select2.scss [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+	&.select2-container--open,
+	&.select2-container--focus {
+
+		.select2-selection--single,
+		.select2-selection--multiple {
+			border-color: $editor-light;
+		}
+	}
+
+	/* Due to the add-new-item button, we need to make make the button vertically aligned to the selected item before it. */
```
Comment (kobizz) [inline assets/dev/scss/editor/_select2.scss:51]:
```suggestion
	/* Due to the add-new-item button, we need to make the button vertically aligned to the selected item before it. */
```

**Review by kobizz — CHANGES_REQUESTED** [review]

**Review by kobizz — APPROVED** [review]

**Review by matipojo — APPROVED** [review]

**Review by KingYes — APPROVED** [review]

**Pull request merged**